insights/parsers/dse_ldif.py

```diff
@@ -0,0 +1,194 @@
+    ...  read,search,compare) userdn="ldap:///anyone";)
+    ... creatorsName: cn=server,cn=plugins,cn=config
+    ... modifiersName: cn=server,cn=plugins,cn=config
+    ... createTimestamp: 20151021124922Z
+    ... modifyTimestamp: 20151021124922Z
+    >>> from insights.tests import context_wrap
+    >>> shared = {dseLdifConfig: dseLdifConfig(context_wrap(dse_ldif_input))}
+    >>> dse_ldif_config = shared[dseLdifConfig]
+    >>> 'objectClass' in dse_ldif_config
+    True
+    >>> 'creatorsName' in dse_ldif_config
+    True
+    >>> dse_ldif_config.get_line('creatorsName')
+    'creatorsName: cn=server,cn=plugins,cn=config'
+    >>> dse_ldif_config.get_values('modifyTimestamp')
+    ['20151021124922Z']
+    >>> dse_ldif_config.last('createTimestamp')
+    '20151021124922Z'    
+"""
+from collections import namedtuple
+from .. import Parser, parser, get_active_lines
+from insights.specs import Specs
+import re
+
+# optional whitespace, at least one non-whitespace (the keyword), at least one whitespace (space), a plus literal, anything
+PLUS_PATTERN = re.compile(r'^\s*\S+\s+\+.*$')
+
+
+@parser(Specs.dse_ldif)
+class dseLdifConfig(Parser):
+    """Parsing for ``/etc/dirsrv/slapd-<>/dse.ldif`` file.
+
+    Properties:
+        lines (list): List of `KeyValue` namedtupules for each line in
+            the configuration file.
+        keywords (set): Set of keywords present in the configuration
+            file, each keyword has been converted to lowercase.
+    """
+
+    KeyValue = namedtuple('KeyValue', ['keyword', 'value', 'kw_lower', 'line'])
+    """namedtuple: Represent name value pair as a namedtuple with case ."""
+
+    # Note about "+" options. In some rhds versions and for some keywords,
+    # a "+" at the beginning of the value list signifies that the values are
+    # added to the default list of values.
+    # IT IS NOT DETECTED AND PARSED HERE.
+    # `parse_content()` doesn't split the individual values because some
+    # keywords don't have a list of values (just one string).
+    # See the docstring in `line_uses_plus()` for more details.
+    # Re: BZ#1697477
+    # Config lines may also be delimited by `=`, and values may be quoted
+    # with `"`. Here it is assumed that config lines are well-formed.
+    def parse_content(self, content):
+        self.lines = []
+        for line in get_active_lines(content):
+            line_splits = [s.strip() for s in re.split(r"[\s:]+", line, 1)]
+            kw, val = line_splits[0], line_splits[1].strip('"') if \
+                len(line_splits) == 2 else ''
+            self.lines.append(self.KeyValue(
+                kw, val, kw.lower(), line
+            ))
+        self.keywords = set([k.kw_lower for k in self.lines])
+
+    def __contains__(self, keyword):
+        """
+        (bool) Is this keyword found in the configuration file?
+        """
+        return keyword.lower() in self.keywords
+
+    def __iter__(self):
```

Comment:
The `__iter__()` method is not tested. Tests are good not only for ensuring the code works as expected, but it is also an always-up-to-date form of documentation.

Reply:
@jsvob Thank you very much for your review. I will be changing the docstring tests after removing shared, as asked by @bfahr then I will add tests for these methods.